lib/postgres_v3_experimental.dart

```diff
@@ -0,0 +1,204 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:collection/collection.dart';
+
+abstract class PgPool implements PgSession, PgSessionExecutor {
```

Comment:
It's unfortunate that Dart pushes us towards namespacing class names (like Obj-C did).  One could go the http route, and just use the good names and expect everyone to import you via import 'package:postgres/postgres.dart' as pg;

Reply:
I like the `package:http`-way too, but in the end it doesn't give much difference to have `pg.X` or `PgX`.

Reply:
Well, the big difference is in the optionality.  Inside package:http they don't have to use http.  And for places where including package:http doesn't conflict with "Client" or "Request" or "get", etc people happily can.

So if you like the package http way, I'd encourage you to consider it here.  You'd delete a ton of code from inside the package. 🤷

Reply:
Rust has gotten me on this pattern.  Rust has strong (and nicer than c++) namespaces and encourages you to just always use nice names for everything.  Obj-C of course (and I guess Dart, sorta?) doesn't, just has a flat namespace, so you were required to come up with your own globally-safe prefix for everything.